### PR TITLE
[FEATURE]: General Optimizations #21

### LINKED ISSUE #21: [FEATURE]: Implement general optimizations
**Is your feature request related to a problem? Please describe.**
>A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
- N/A

**Describe the solution you'd like**
>A clear and concise description of what you want to happen.
- Workflow to uniformly format the entire codebase, instead of relying on individual linting
- Refactor to reflect the rest of the repository's naming convention

**Describe alternatives you've considered**
>A clear and concise description of any alternative solutions or features you've considered.
- N/A

**Additional context**
>Add any other context or screenshots about the feature request here.
- Finding a simple and effective way to organize the codebase

### PR DESCRIPTION
**Pull request by HarryVasanth**
### Related issues

Closes #21

### What does this PR do?

> <sub>Clearly state the reason for this change. What problem is it solving or feature is it adding?</sub>
- Prettified codebase
- Refactored codebase
- Added standardized linting and formatting

### Solution Overview

> <sub>Provide an overview of the solution implemented in this pull request. This should be a high level overview without getting into technical details. If applicable, include screenshots of UI or use GitHub compliant [`mermaid`](https://mermaid.live/) graphs to visually represent the solution.</sub>

- Added workflow to uniformly format the entire codebase, instead of relying on individual linting
- Refactored to reflect the rest of the repository's naming convention

### Implementation Details

> <sub>Explain the details of the implementation and the reasoning behind it. Where should the reviewer start?</sub>

- This is the simple yet effective way to keep the codebase organized

### CODE REVIEW

**Comment by Dntfreitas** [comment]
LGTM

File: .github/workflows/format_codebase.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: 💅 Format Codebase
+
+on:
```
Comment (Dntfreitas) [inline .github/workflows/format_codebase.yml:3]: Is the workflow triggered when a CSV or a JSON file is changed? Would we need to approve a PR every week if that's the case?

Would you consider changing the trigger to:
```json
on:
  push:
    paths-ignore:
      - '**.json'
      - '**.csv'
```

**Review by Dntfreitas — APPROVED** [review]

**Review by 13dev — APPROVED** [review]
great job

File: .github/workflows/format_codebase.yml (hunk 1 above)
Comment (HarryVasanth) [reply .github/workflows/format_codebase.yml:3]: No, the prettier would run automatically, only the black formatter (for python files) opens a PR.

Comment (Dntfreitas) [reply .github/workflows/format_codebase.yml:3]: That's perfect!

**Review by carlosrsabreu — COMMENTED** [review]
Looks awesome!! Nice work!

Just a note: is there a way to create a lint file in this project so these rules would be applied when someone changes and saves the file locally? This way, the number of PRs to accept and the chances of someone having the code formatted in the "wrong" way would be less. Let me know your thoughts, @HarryVasanth

**Comment by HarryVasanth** [comment]
> Looks awesome!! Nice work!
> 
> Just a note: is there a way to create a lint file in this project so these rules would be applied when someone changes and saves the file locally? This way, the number of PRs to accept and the chances of someone having the code formatted in the "wrong" way would be less. Let me know your thoughts, @HarryVasanth

There are no specific lint config for this. For `py`, literally the whole world uses [**Black**](https://github.com/psf/black#used-by) and it is available as a linter in all the mainstream code editors. The `json` and `md` files are formatted using [Prettier](https://prettier.io/) which has more than 5.2 million dependant projects.

**Comment by HarryVasanth** [comment]
After speaking with @carlosrsabreu, I am removing the additional step of approving the `pull-request` created by the black formatter. From now on, all the file types in the workflow (json, md, py) that are pushed to the `main` branch will be formatted accordingly.

**Pull request merged**